kafka/src/test/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactoryClusteringTest.java

```diff
@@ -0,0 +1,112 @@
+package org.axonframework.extensions.kafka.eventhandling.producer;
+
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.axonframework.extensions.kafka.eventhandling.util.KafkaAdminUtils;
+import org.axonframework.extensions.kafka.eventhandling.util.KafkaContainerClusterTest;
+import org.junit.jupiter.api.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.IntStream;
+
+import static org.axonframework.extensions.kafka.eventhandling.producer.DefaultProducerFactory.builder;
+import static org.axonframework.extensions.kafka.eventhandling.util.ProducerConfigUtil.minimalTransactional;
+import static org.axonframework.extensions.kafka.eventhandling.util.ProducerConfigUtil.transactionalProducerFactory;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Tests for {@link DefaultProducerFactory} when running a cluster.
+ *
+ * @author Nakul Mishra
+ * @author Steven van Beelen
+ * @author Lucas Campos
+ */
+
+public class DefaultProducerFactoryClusteringTest extends KafkaContainerClusterTest {
+
+    private static final String TOPIC = "testSendingMessagesUsingMultipleTransactionalProducers";
+
+    @BeforeAll
+    static void before() {
+        KafkaAdminUtils.createTopics(KAFKA_CLUSTER.getBootstrapServers(), TOPIC);
```

Comment:
Nit: `KAFKA_CLUSTER.getBootstrapServers()` can be moved to `KafkaContainerClusterTest` as a base method that the tests can use without knowing about the Container or Cluster at all